packages/maskbook/src/extension/background-script/HelperService.ts

```diff
@@ -26,12 +27,25 @@ export function fetch(url: string) {
     return globalThis.fetch(url).then((x) => x.blob())
 }
 
-export function saveAsFile(file: ArrayBuffer, mineType: string, suggestingFileName: string) {
-    const blob = new Blob([file], { type: mineType })
+const createDlElement = once(() => document.createElement('a'))
+
+export function saveAsFileFromUrl(url: string, fileName = '') {
+    if (process.env.architecture === 'app' && process.env.firefoxVariant === 'geckoview') {
+        const dlElem = createDlElement()
+        dlElem.href = url
+        dlElem.download = fileName
+        dlElem.click()
+    } else {
+        browser.downloads.download({
+            url,
+            filename: fileName,
+            saveAs: true,
```

Comment:
IIRC here is some problem with "saveAs: true" in some platforms

Reply:
Is that Android?

Reply:
@zhouhanseng just have a test on Chrome, Firefox and iOS to confirm

Reply:
The only platform I can't test is Android, it works on Chrome, Firefox and iOS.

Reply:
Tested on Android, still unable to trigger downloading.
Since this PR fixes image download on iOS, I suggest merge this PR for iOS release, and leave the corresponding Android issue open